daemon/api.go

```diff
@@ -758,13 +775,14 @@ func storeUpdates(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	state := c.d.overlord.State()
 	state.Lock()
+	defer state.Unlock()
```

Comment:
Most of the other code uses a pattern of lock/meta/unlock - this does not. Might be worthwhile to point out why it is different here. It also means that the lock is held until sendStorePackages has finished sending. But it seems we do this in a lot of places so not relevant for this PR.